File: Lab01/Python/lab01.py
'''Find Min and Max numbers'''

data = [90, 40, 60, 10, 50, 80, 20, 70, 15, 25]

# Python
max_cost = max(data)
min_cost = min(data)

print('Minimum number is ', min_cost, 'at position ', data.index(min_cost) + 1)
print('Maximum number is ', max_cost, 'at position ', data.index(max_cost) + 1)

# C++

def findMin(data, n):
    min = data[0]
    for i in range(1, n):
        if data[i] < min:
            min = data[i]
    return min

def findMax(data, n):
    max = data[0]
    for i in range(1, n):
        if data[i] > max:
            max = data[i]
    return max

max_cst = findMax(data, len(data))
min_cst = findMin(data, len(data))

print('Minimum number is ', min_cst, 'at position ', data.index(min_cst) + 1)
print('Maximum number is ', max_cst, 'at position ', data.index(max_cst) + 1)

File: Lab01/C++/Lab01.cpp
#include <iostream>
#include <fstream>
#include <string>
#include <vector>
using namespace std;

//최댓값 최소값 찾아내서 몇 번 째 위치에 있는지 출력하기

int findMax(vector<int>data, int n)
{
	int max = data[0];
	int maxPos = 0;
	for (int i = 1; i < n; i++)
	{
		if (data[i]>max)
		{
			max = data[i];
			maxPos = i;
		}
	}
	return maxPos;
}

int findMin(vector<int>data, int n)
{
	int min = data[0];
	int minPos = 0;
	for (int i = 1; i < n; i++)
	{
		if (data[i] < min)
		{
			min = data[i];
			minPos = i;
		}
	}
	return minPos;
}

int main(void)
{
	vector<int>vec;
	ifstream in("Lab0-1.dat");

	while (!in.eof())
	{
		int val;
		in >> val;
		vec.push_back(val);
	}

	int maxPos = findMax(vec, vec.size());
	int minPos = findMin(vec, vec.size());

	cout << "Minumum number is " << vec[minPos] << " at position " << minPos << ',' << endl;
	cout<< "Maximum number is " << vec[maxPos] << " at position " << maxPos << endl;
}